csrc/host_ir/pass/deallocate.cpp

```diff
@@ -0,0 +1,52 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <host_ir/pass/deallocate.h>
+
+namespace nvfuser::hir {
+
+void insertDeallocations(HostIrContainer* hic) {
+  const std::vector<Expr*>& topLevelExprs = hic->topLevelExprs();
+  std::unordered_map<TensorView*, std::vector<nvfuser::Expr*>::size_type>
+      lastUse;
+  for (std::vector<nvfuser::Expr*>::size_type i = 0; i < topLevelExprs.size();
+       i++) {
+    Expr* expr = topLevelExprs[i];
+    if (expr->isA<Deallocate>()) {
```

Comment:
Do you expect the input host IR to contain Deallocates already? I don't see a good reason for that to happen, and I'd probably just NVF_ERROR that the input can't have Deallocate and simplify.